test/internal/packagemanifests.go

```diff
@@ -65,8 +66,12 @@ func (tc TestContext) AddPackagemanifestsTarget() error {
 	if !strings.HasPrefix(c.Layout, "go") {
 		// The target is equals for all types but NonGo projects has not the manifests target which
 		// needs to be replaced here.
+		manifestsTarget := "packagemanifests: kustomize manifests"
+		if !strings.Contains(makefilePackagemanifestsFragment, manifestsTarget) {
+			return errors.New("unable to find the manifests target to be replaced")
+		}
 		makefilePackagemanifestsFragment = strings.Replace(makefilePackagemanifestsFragment,
-			"packagemanifests: kustomize manifests", "packagemanifests: kustomize", 1)
+			manifestsTarget, "packagemanifests: kustomize", 1)
```

Comment:
Nit: string comparison and replacement seems more complex than the `fmt.Sprintf` idea, and it is still prone to accidental breakage. At least now the test would fail, but it seems more efficient and less error prone to build the string instead of search/replace it. 

With this, it is possible that we'll make a change and not see that we accidentally broke something until 20-30 minutes into a CI run.